portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,5 +43,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   }
 
 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    ArrayList<String> messages = new ArrayList<String>();
+    String comment = getParameter(request, "textInput", "nothing");
+    messages.add(comment);
+
+    Gson gson = new Gson();
+    String json = gson.toJson(messages);
 
+    response.setContentType("text/html;");
+    response.getWriter().println(json);
```

Comment:
No need to put the doGet() stuff here. Please remove and readd `response.sendRedirect("/index.html");`. You need to tell the page to redirect there, so the UI can be rerendered and doGet() method gets triggered.